src/terrain/routes.clj

```diff
@@ -64,6 +64,11 @@
         (tc/with-logging-context {:user-info (cheshire/encode user-info)}
           (handler request))))))
 
+(defn optionally-authenticated-routes
+  []
+  (util/flagged-routes
+   (secured-filesystem-routes)))
```

Comment:
Also note that `optionally-authenticated-routes` has no call to `route/not-found`. This allows the request to fall through to `secured-routes-no-context` if none of the optionally authenticated routes match.

Reply:
Good point. I'll add a docstring.